### PR TITLE
JSON and serialization edits

### PR DESCRIPTION
**Pull request by pedromorgan**
renamed  to avoid confuse, and small tweaks..

### CODE REVIEW

**Comment by pedromorgan** [comment]
I signed it! cymro bach yn cymru

**Comment by chalin** [comment]
Thanks for your PR, but use of `json` as the formal parameter name is the convention (idiom?) followed when using, https://pub.dartlang.org/packages/json_serializable based code.

@kevmoo - do you agree with the statement above, or have any other comments about this?

cc @kwalrath

**Comment by kevmoo** [comment]
> Thanks for your PR, but use of json as the formal parameter name is the convention (idiom?) followed when using, https://pub.dartlang.org/packages/json_serializable based code.

SGTM

File: src/docs/development/data-and-backend/json.md [hunk 1]
```diff
@@ -127,13 +130,13 @@ with the JSON string as the method argument.
 
 <!-- skip -->
 ```dart
-Map<String, dynamic> user = json.decode(json);
+Map<String, dynamic> user = json.decode(user_data_str);
```
Comment (chalin) [inline src/docs/development/data-and-backend/json.md]: @kwalrath @kevmoo - should we be promoting use of `jsonDecode()` here instead? /cc @Hixie @sfshaza2

**Comment by chalin** [comment]
After some discussion with @kwalrath, I'm warming up to some of the changes made in this PR. Let's see if we can make it work.

File: src/docs/development/data-and-backend/json.md [hunk 2]
```diff
@@ -78,10 +78,13 @@ encoding, see
...
-Such a library would require using runtime reflection, which is disabled in
-Flutter. Runtime reflection interferes with _tree shaking_, which Dart has
+Such a library would require using runtime 
+[reflection](https://en.wikipedia.org/wiki/Reflection_(computer_programming)) 
+which is disabled in
+Flutter. Runtime reflection interferes with 
+[tree shaking](https://en.wikipedia.org/wiki/Tree_shaking), which Dart has
 supported for quite a long time. With tree shaking, you can “shake off” unused
-code from your release builds. This optimizes the app's size significantly.
+code from the release builds and this optimizes the app's size significantly.
```
Comment (chalin) [inline src/docs/development/data-and-backend/json.md]: @kwalrath - are you ok with the changes to this paragraph?

File: src/docs/development/data-and-backend/json.md [hunk 3]
```diff
@@ -184,7 +188,7 @@ itself. With this new approach, you can decode a user easily.
 
 <!-- skip -->
 ```dart
-Map userMap = json.decode(json);
+Map userMap = json.decode(string_with_json);
```
Comment (chalin) [inline src/docs/development/data-and-backend/json.md]: In the [Library Tour](https://www.dartlang.org/guides/libraries/library-tour#dartconvert---decoding-and-encoding-json-utf-8-and-more), we use the variable name `jsonString`. Could you change it to that name?

File: src/docs/development/data-and-backend/json.md [hunk 4]
```diff
@@ -197,7 +201,7 @@ already does it for you.
 
 <!-- skip -->
 ```dart
-String json = json.encode(user);
+String data_string = json.encode(user);
```
Comment (chalin) [inline src/docs/development/data-and-backend/json.md]: `data_string` -> `jsonString`

File: src/docs/development/data-and-backend/json.md (hunk 2 above)
Comment (kwalrath) [reply src/docs/development/data-and-backend/json.md]: I like the original text better.

File: src/docs/development/data-and-backend/json.md (hunk 1 above)
Comment (kwalrath) [reply src/docs/development/data-and-backend/json.md]: I personally like `jsonDecode()`, but not everyone agrees. @kevmoo?

File: src/docs/development/data-and-backend/json.md (hunk 2 above)
Comment (chalin) [reply src/docs/development/data-and-backend/json.md]: How about the added links to Wikipedia?

Comment (kwalrath) [reply src/docs/development/data-and-backend/json.md]: Ah, missed those. I like them, but we should move the punctuation inside the brackets, so the external link markers don't look weird. So:

[reflection] -> [reflection,]
[tree shaking](...), -> [treeshaking,](...)

File: src/docs/development/data-and-backend/json.md [hunk 5]
```diff
@@ -167,9 +171,9 @@ class User {
 
   User(this.name, this.email);
 
-  User.fromJson(Map<String, dynamic> json)
-      : name = json['name'],
-        email = json['email'];
+  User.fromJson(Map<String, dynamic> data_map)
+      : name = data_map['name'],
+        email = data_map['email'];
```
Comment (chalin) [inline src/docs/development/data-and-backend/json.md]: Use of `json` here is ok. Revert these changes.

File: src/docs/development/data-and-backend/json.md [hunk 6]
```diff
@@ -151,6 +154,7 @@ class, called `User` in this example. Inside the `User` class, you'll find:
 * A `User.fromJson` constructor, for constructing a new `User` instance from a
   map structure.
 * A `toJson` method, which converts a `User` instance into a map.
+* Note: the `from*` `to*` naming is recommended but optional eg create a `fromMyFormat` etc
```
Comment (chalin) [inline src/docs/development/data-and-backend/json.md]: I'd suggest dropping this note. @kwalrath - WDYT?

File: src/docs/development/data-and-backend/json.md (hunk 2 above)
Comment (chalin) [reply src/docs/development/data-and-backend/json.md]: Also, move the link definitions to the end of the file, rather than inlining them. E.g.,

    Such a library would require using runtime [reflection][] ...
    
    [reflection]: https://en.wikipedia.org/wiki/Reflection_(computer_programming)

Comment (kwalrath) [reply src/docs/development/data-and-backend/json.md]: Of course, if you have the commas in there, you'll probably want to use something like `[reflection,][reflection]`.

Comment (chalin) [reply src/docs/development/data-and-backend/json.md]: Oops, yes, thanks.

**Comment by pedromorgan** [comment]
@chalin NO.. why not do edits yourself. make a decision, and rip the good parts, ignore others ;-))

One thing I dislike is the `your can, we can one can.`
cos this is crap when u speak+understand welsh, japanese and english.. doth not translate...

The language = manual needs to me more "commanding"..
```you can create a new app by.....```
to
```Create a new app, by...```

File: src/docs/development/data-and-backend/json.md (hunk 1 above)
Comment (kevmoo) [reply src/docs/development/data-and-backend/json.md]: Ditto! Avoid confusion w/ local var/param named `json`

Comment (chalin) [reply src/docs/development/data-and-backend/json.md]: FYI, I've created: _Code cleanup: `json.decode()` -> `jsonDecode()`_ #2021
